http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala

```diff
@@ -214,7 +214,7 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
   def checkAgain(
     monitorContext: MonitorContext,
     runtimeAndRuntimeConfig: RuntimeAndRuntimeConfig,
-    dataprocInstances: Set[DataprocInstance], //only applies to dataproc
+    fetchDataprocInstances: Option[F[Set[DataprocInstance]]], //only applies to dataproc
```

Comment:
made this into an `Option[F[Set[DataprocInstance]]]` so that we don't have to eagerly fetch instances at call site